text/0000-otlp-trace-data-format.md

```diff
@@ -0,0 +1,383 @@
+# OTLP Trace Data Format
+
+_Author: Tigran Najaryan, Omnition Inc._
+
+**Status:** `proposed`
+
+OTLP Trace Data Format specification describes the structure of the trace data that is transported by OpenTelemetry Protocol (RFCNNN).
+
+## Motivation
+
+This document is a continuation of OpenTelemetry Protocol RFCNNN and is necessary part of OTLP specification.
+
+## Explanation
+
+OTLP Trace Data Format is primarily inherited from OpenCensus protocol. Several changes are introduced with the goal of more efficient serialization. Notable differences from OpenCensus protocol are:
+
+1. Removed `Node` as a concept.
+2. Extended `Resource` to better describe the source of the telemetry data.
+3. Replaced attribute maps by lists of key/value pairs.
+4. Eliminated unnecessary additional nesting in various values.
+
+Changes 1-2 are conceptual, changes 3-4 improve performance.
+
+## Internal details
+
+This section specifies data format in Protocol Buffers.
+
+### Resource
+
+```
+// Resource information. This describes the source of telemetry data.
+message Resource {
+  // Identifier that uniquely identifies a process within a VM/container.
+  Process process = 1;
+
+  // Name of the service.
+  string service_name = 2;
+
+  // Set of labels that describe the resource.
+  repeated AttributeKeyValue attributes = 3;
```

Comment:
With this change you add support for multiple types of values for the attributes in the Resource. Can you clarify the need?

Reply:
I think it can be very useful. For example let's imagine I want to have process ID in the resource. Yes, I can also use strings for a PID but it is typically numeric in modern OS-es so using the right type will not hurt.
This is only very slightly less efficient than using string-only labels and I think it is worth it.